quic/s2n-quic-core/src/recovery/hybrid_slow_start.rs

```diff
@@ -94,14 +113,39 @@ impl HybridSlowStart {
         if let (N_SAMPLING, Some(last_min_rtt), Some(cur_min_rtt)) =
             (self.sample_count, self.last_min_rtt, self.cur_min_rtt)
         {
-            let threshold = last_min_rtt / THRESHOLD_DIVIDEND;
-            // Clamp n to the min and max thresholds
-            let threshold = threshold.min(MAX_DELAY_THRESHOLD).max(MIN_DELAY_THRESHOLD);
-            let delay_increase_is_over_threshold = cur_min_rtt >= last_min_rtt + threshold;
-            let congestion_window_is_above_minimum = congestion_window >= self.low_ssthresh();
-
-            if delay_increase_is_over_threshold && congestion_window_is_above_minimum {
-                self.threshold = congestion_window;
+            if congestion_window >= self.cssthreshold {
```

Comment:
this is getting a little confusing with all the ifs and thresholds, etc. I'm wondering if it might be easier to follow with an explicit `State` enum such as:
```rust
enum State {
     // Determining css threshold
    CONSERVATIVE_SLOW_START(f32),
    // Determining the slow start threshold
    SLOW_START,
    // The overall slow start threshold has been determined
    THRESHOLD_DETERMINED,
}
```
might not be exactly like this, but I think it might be worthwhile to think about now that slow start is more than just a single threshold

Reply:
A tricky thing for this is we're currently keeping the current logic in the code. But, the current logic applies hystart algorithm even after timeout. So, the logic doesn't need THRESHOLD_DETERMINED state while hystart++ logic needs it. 
I am thinking that it might be better to do this after we fully migrate to hystart++.